Sming/Core/Network/MqttClient.cpp

```diff
@@ -286,13 +280,13 @@ bool MqttClient::subscribe(const String& topic)
 		return false;
 	}
 
-	mqtt_message_t* message = (mqtt_message_t*)malloc(sizeof(mqtt_message_t));
-	mqtt_message_init(message);
-	message->common.type = MQTT_TYPE_SUBSCRIBE;
-	message->subscribe.topics = (mqtt_topicpair_t*)malloc(sizeof(mqtt_topicpair_t));
-	memset(message->subscribe.topics, 0, sizeof(mqtt_topicpair_t));
+	mqtt_message_t message;
```

Comment:
You are creating here a local variable and then you are passing the address of that local variable to be used later from the requestQueue. I am not sure this is a good idea. By the time the message is dequeued the memory address will most certainly point to something unexpected.

Reply:
I've taken another go at this, much happier with it.

Reply:
> I've taken another go at this, much happier with it.

Now it looks much better :)

Reply:
On the Host it works like charm.

```
55018194 MQTT replacing connect message
MQTT Broker Disconnected!!
Connecting to 	mqtt://attachix.zom:1883/
57112580 MQTT replacing connect message
MQTT Broker Disconnected!!
Connecting to 	mqtt://attachix.zom:1883/
59217835 MQTT replacing connect message
MQTT Broker Disconnected!!
Connecting to 	mqtt://attachix.zom:1883/
61317774 MQTT replacing connect message
MQTT Broker Disconnected!!
...
```
and the final summary
```
==5606== LEAK SUMMARY:
==5606==    definitely lost: 0 bytes in 0 blocks
==5606==    indirectly lost: 0 bytes in 0 blocks
==5606==      possibly lost: 0 bytes in 0 blocks
```